packages/flutter/test/rendering/mock_canvas.dart

```diff
@@ -1003,19 +1003,20 @@ class _CirclePaintPredicate extends _DrawCommandPaintPredicate {
   void verifyArguments(List<dynamic> arguments) {
     super.verifyArguments(arguments);
     final Offset pointArgument = arguments[0];
+
     if (x != null && y != null) {
       final Offset point = new Offset(x, y);
       if (point != pointArgument)
-        throw 'It called $methodName with a center coordinate, $pointArgument, which was not exactly the expected coordinate ($point).';
+        throw 'It called $methodName with a center coordinate, Offset(${pointArgument.dx}, ${pointArgument.dy}), which was not exactly the expected coordinate (Offset(${point.dx}, ${point.dy})).';
     } else {
       if (x != null && pointArgument.dx != x)
-        throw 'It called $methodName with a center coordinate, $pointArgument, whose x-coordinate not exactly the expected coordinate (${x.toStringAsFixed(1)}).';
+        throw 'It called $methodName with a center coordinate, Offset(${pointArgument.dx}, ${pointArgument.dy}), whose x-coordinate not exactly the expected coordinate ($x).';
       if (y != null && pointArgument.dy != y)
-        throw 'It called $methodName with a center coordinate, $pointArgument, whose y-coordinate not exactly the expected coordinate (${y.toStringAsFixed(1)}).';
+        throw 'It called $methodName with a center coordinate, Offset(${pointArgument.dx}, ${pointArgument.dy}), whose y-coordinate not exactly the expected coordinate ($y).';
     }
     final double radiusArgument = arguments[1];
     if (radius != null && radiusArgument != radius)
-      throw 'It called $methodName with radius, ${radiusArgument.toStringAsFixed(1)}, which was not exactly the expected radius (${radius.toStringAsFixed(1)}).';
+      throw 'It called $methodName with radius, $radiusArgument, which was not exactly the expected radius ($radius).';
```

Comment:
we really should only do these when it matters, because the output is really ugly when it doesn't, but yeah.

Reply:
Maybe we need a function that diffs two doubles, and returns identical strings if they are identical, and different strings when they are different?  Like so:

```dart
List<String> diffDouble(double a, double b) {
  if (a == b || (a * 10.0).truncate() != (b * 10.0).truncate()) {
    return <String>[a.toStringAsFixed(1), b.toStringAsFixed(1)];
  } else {
    String aStr = a.toStringAsFixed(15);
    String bStr = b.toStringAsFixed(15);
    int digits = a.toStringAsFixed(1).length;
    while (digits <= aStr.length && aStr[digits] == bStr[digits]) {
      digits++;
    }
    if (digits < aStr.length) {
    	digits++;
    }
    return <String>[aStr.substring(0, digits), bStr.substring(0, digits)];
  }
}
```

diffDouble(1.0, 2.0) => ['1.0', '2.0']
diffDouble(1.0, 1.0000012) -> ['1.000000', '1.000001']

Reply:
yeah, that's be great for tests